pkg/prowgen/prowgen.go

```diff
@@ -618,6 +619,10 @@ func generateJobBase(name, prefix string, info *ProwgenInfo, podSpec *corev1.Pod
 	} else if !skipCloning && info.Config.Private {
 		decorationConfig = &prowv1.DecorationConfig{OauthTokenSecret: &prowv1.OauthTokenSecret{Key: api.OauthTokenSecretKey, Name: api.OauthTokenSecretName}}
 	}
+	maxCustomDuration := time.Hour * 8
+	if timeout != nil && timeout.Duration <= maxCustomDuration {
+		decorationConfig.Timeout = timeout
+	}
```

Comment:
I think we should actively refuse this in ci-op config validation (https://github.com/openshift/ci-tools/blob/master/pkg/validation/test.go#L116) with some sensible error message instead of capping it silently.

Reply:
good idea, implemented